### PR TITLE
FOGL-3975 Add documentation

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+.. |fledge-filter-threshold| raw:: html
+
+   <a href="../fledge-filter-threshold/index.html">fledge-filter-threshold</a>
+
+
+.. |fledge-filter-expression| raw:: html
+
+   <a href="../fledge-filter-expression/index.html">fledge-filter-expression</a>
+
+Rate Control Filter
```
Comment (ashish-jabble) [inline docs/index.rst]: Rate Filter

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    - **End Expression**: An expression to clear the trigger for full rate data, if left blank this will simply be the trigger filter evaluating to false
+
+    - **Full rate time (ms)**: The time window, in milliseconds to forward data at the full rate
+
+    - **Pre-trigger time (ms)**: An optional pre-trigger time expressed in milliseconds
+
+    - **Reduced collection rate**: The nominal data rate to send data out. This defines the period over which is outgoing data item is averaged.
+
+    - **Rate Units**: The units that the reduced collection rate is expressed in; per second, minute or hour
```
Comment (ashish-jabble) [inline docs/index.rst]: or day

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+The trigger expression uses the same expression mechanism, |exprtk|  as the
+|fledge-south-expression|, |fledge-filter-expression| and |fledge-filter-threshold| plugins
+
+Expression may contain any of the following...
+
+  - Mathematical operators (+, -, \*, /, %, ^)
+
+  - Functions (min, max, avg, sum, abs, ceil, floor, round, roundn, exp, log, log10, logn, pow, root, sqrt, clamp, 
+inrange, swap)
```
Comment (ashish-jabble) [inline docs/index.rst]: Block quote ends without a blank line; unexpected unindent.

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+
+.. |fledge-filter-expression| raw:: html
+
+   <a href="../fledge-filter-expression/index.html">fledge-filter-expression</a>
+
+Rate Control Filter
+===================
+
+A filter plugin that can be used to reduce the rate a reading is stored
```
Comment (ashish-jabble) [inline docs/index.rst]: A `fledge-filter-rate` plugin is a filter that ...

**Review by ashish-jabble — CHANGES_REQUESTED** [review]

File: docs/index.rst (hunk 4 above)
Comment (MarkRiddoch) [reply docs/index.rst]: I got bored using the same words!

**Review by ashish-jabble — APPROVED** [review]
Thanks for the changes.

**Pull request merged**